### PR TITLE
Gson improvements

### PR DESCRIPTION
**Pull request by triplem**
Like discussed, this implements the same DSL as already used with Jackson. Furthermore this implements the changes necessary for #231.

### CODE REVIEW

File: ktor-samples/ktor-samples-gson/src/io/ktor/samples/gson/GsonApplication.kt [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
 import io.ktor.response.*
 import io.ktor.routing.*
+import java.text.*
 
 data class Model(val name: String, val items: List<Item>)
 data class Item(val key: String, val value: String)
 
 /*
-         > curl -v --compress --header "Accept: application/json" http://localhost:8080/v1
+         > curl -v --compress --header "Accept: application/gson" http://localhost:8080/v1
```
Comment (orangy) [inline ktor-samples/ktor-samples-gson/src/io/ktor/samples/gson/GsonApplication.kt]: Nope, here stays "json" :)

File: ktor-samples/ktor-samples-gson/src/io/ktor/samples/gson/GsonApplication.kt [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
          {"name":"root","items":[{"key":"A","value":"Apache"},{"key":"B","value":"Bing"}]}
+         The result is pretty printed, to show off how to configure gson, but it is
+         possible to use the default gson as well
 
-         > curl -v --compress --header "Accept: application/json" http://localhost:8080/v1/item/A
+         > curl -v --compress --header "Accept: application/gson" http://localhost:8080/v1/item/A
```
Comment (orangy) [inline ktor-samples/ktor-samples-gson/src/io/ktor/samples/gson/GsonApplication.kt]: And here

**Comment by orangy** [comment]
Otherwise good. Please fix and I'll merge.

**Comment by cy6erGn0m** [comment]
Merged manually. Auto-closed because of removed `develop` branch. Sorry for inconvenience

**Pull request closed without merging**